OTAnalytics/domain/track.py

```diff
@@ -459,24 +711,24 @@ def get_for(self, id: TrackId) -> Optional[Track]:
         Returns:
             Optional[Track]: track if it exists
         """
-        return self._tracks.get(id)
+        return self._dataset.get_for(id)
 
-    def get_all(self) -> list[Track]:
+    def get_all(self) -> TrackDataset:
         """
         Retrieve all tracks.
 
         Returns:
             list[Track]: all tracks within the repository
         """
-        return list(self._tracks.values())
+        return self._dataset
 
     def get_all_ids(self) -> Iterable[TrackId]:
         """Get all track ids in this repository.
 
         Returns:
             Iterable[TrackId]: the track ids.
         """
-        return self._tracks.keys()
+        return self._dataset.get_all_ids()
 
     def remove(self, track_id: TrackId) -> None:
```

Comment:
That is true. The `TrackRemoveError` should be thrown by the `TrackDataset`-implementation. In case of the pandas-DataFrame, if a key is not available, the remove will not fail, because we just filter all rows. Should we raise an error in such a case?

The result of the operation is correct, the ID to remove does no longer exist in the Dataset.